include/seqan3/search/detail/policy_max_error.hpp

```diff
@@ -24,44 +25,90 @@ namespace seqan3::detail
 //!\ingroup search
 struct policy_max_error
 {
-protected:
+public:
     /*!\brief Returns a detail::search_param object filled by the information from the configuration.
      * \tparam configuration_t The search configuration type.
-     * \tparam query_t Must model std::ranges::foward_range over the index's alphabet.
-     * \param[in] cfg The configuration object.
+     * \tparam query_t         Must model std::ranges::forward_range over the index's alphabet.
+     * \param[in] cfg   The configuration object.
      * \param[in] query The current query sequence.
-     *
-     * \details
-     * 
-     * If seqan3::max_error is set in the configuration, its value already is of type detail::search_param and
-     * can be returned as is.
-     * If seqan3::max_error_rate is set in the configuration, the error rates are converted to error counts
-     * based on the length of the query sequence.
      */
     template <typename configuration_t, std::ranges::forward_range query_t>
-    auto max_error_counts(configuration_t const & cfg, [[maybe_unused]] query_t && query)
+    auto max_error_counts(configuration_t const & cfg, query_t && query)
     {
         using search_traits_t = search_traits<configuration_t>;
 
-        // retrieve error numbers / rates
-        detail::search_param max_error{0, 0, 0, 0};
+        detail::search_param errors{0, 0, 0, 0}; // total, substitution, insertion, deletion
 
-        if constexpr (search_traits_t::search_with_max_error)
+        [[maybe_unused]] auto query_size = std::ranges::size(query);
```

Comment:
The `query_size` is always used, no?

Reply:
No because the else case has if cases again. If no error is set, we go into the else case and nothing is happening there.
```bash
/Users/lydia/Repos/seqan3/include/seqan3/search/detail/policy_max_error.hpp:44:14: error: variable 'query_size' set but not used [-Werror=unused-but-set-variable]
   44 |         auto query_size = std::ranges::size(query);
      |              ^~~~~~~~~~

```